WordPress/gutenberg
# Social Link: Add border and spacing support

## What?

Add border block support to the `Social Links` block.

Part of,
https://github.com/WordPress/gutenberg/issues/43247
https://github.com/WordPress/gutenberg/issues/43243

## Why?
`Social Link` block is missing border and spacing support.

## How?

Adds the border block support in block.json

## Testing Instructions

- Go to Global Styles setting ( under appearance > editor > styles > edit styles > blocks )
- Make sure that `Social Link` styles are configurable via Global Styles
- Verify that Global Styles are applied correctly in the editor and frontend
- Edit template/page, Add `Social Link` block and Apply the styles
- Verify that block styles take precedence over global styles
- Verify that block styles display correctly in both the editor and frontend

## Screenshots or screencast
